Support multiple shells

wm opened the pull request:
I wanted to use this for my zsh shell so I updated it to allow multiple shells.

I also extracted most segment functionality into functions with an eye toward making this customizable.

Finally I updated the README with usage instructions.

wm committed:
Convert to zsh

wm committed:
fix escaping

wm committed:
extraction

wm committed:
Support for different shells

wm committed:
Add usage instructions to README

wm committed:
Update the gitignore

wm committed:
Add hostname if ssh; cross compile with docker.

sivel commented on powerline-shell.go:
```diff
@@ -25,6 +24,8 @@ import (
 	"regexp"
 	"strings"
+
+	"github.com/wm/powerline-shell-go/powerline"
```
This should reference the parent repo and not your fork.

wm replied:
Addressed in e00204c

sivel commented:
There seems to be a change in output between master and this PR.  The initial path tested was is `~/go/src/github.com/sivel/powerline-shell-go`

Before:
![screenshot 2015-01-05 at 4 23 07 pm](https://cloud.githubusercontent.com/assets/39340/5621062/5b5c35fa-94f7-11e4-8124-165345356f41.png)

After:
![screenshot 2015-01-05 at 4 22 32 pm](https://cloud.githubusercontent.com/assets/39340/5621063/5b709590-94f7-11e4-9858-db887b19f838.png)

You will notice the omission of the initial `go` from the path.

Also with `/private/tmp`:

Before:
![screenshot 2015-01-05 at 4 28 05 pm](https://cloud.githubusercontent.com/assets/39340/5621118/e68ea6c6-94f7-11e4-9909-c0e35f518561.png)

After:
![screenshot 2015-01-05 at 4 27 47 pm](https://cloud.githubusercontent.com/assets/39340/5621119/e6a2dd30-94f7-11e4-8b5e-86337152c607.png)

You will notice the addition of `/` at the front that shouldn't be there.  (Note that there may have been an initial bug for this last test, as it should have not had the `...` segment there.

wm committed:
Fix cwd output; adds tests

wm commented:
I fixed the cwd segments (in ca3f519) to not output the / unless you are in root. I fixed the bug you mentioned you had too (private > ... > tmp). 

I added some tests for addCwd since it is hard to understand what is happening when. I think the tests are straightforward though.

thanks for the feedback

wm committed:
More tests

sivel commented:
Great, I'll look into testing the changes soon.

sivel commented:
Looking better.  When a python virtual environment is present/activated, the venv should appear at the start.

This image shows the output from your PR first, and the original second.

![screenshot 2015-01-06 at 11 35 20 am](https://cloud.githubusercontent.com/assets/39340/5632917/2ba05246-9598-11e4-82da-b798aba2fa12.png)

wm committed:
Move virtual machine to beginning of prompt

wm committed:
change colors for ssh hostname

wm commented:
okay, I moved the virtual machine to the beginning.

sivel commented:
This was manually merged a long time ago.  Not sure why I never closed the PR.
